framework/wazuh/core/utils.py

```diff
@@ -2005,10 +2024,11 @@ def validate_wazuh_xml(content: str, config_file: bool = False):
         # Check if remote commands are allowed if it is a configuration file
         if config_file:
             check_remote_commands(final_xml)
+            check_agents_allow_higher_versions(final_xml)
+            check_indexer(final_xml)
```

Comment:
Moved them above `check_wazuh_limits_unchanged` to avoid reading the file and fail early if the other sections contain invalid data.